src/modules/auth/helper/fabricWallet.ts

```diff
@@ -81,13 +77,43 @@ export class FabricWallet {
     // if affiliation is specified by client, the affiliation value must be configured in CA
     const secret = await caClient.register(
       {
-        affiliation: orgName + '.' + affiliation,
+        affiliation: orgName,
         enrollmentID: username,
-        role: 'client',
+        role,
+        attrs:
+          role === 'client'
```

Comment:
Change this the otherway around where the admin sends the array, and client sends empty array.